### PR TITLE
Af1.0

### PR DESCRIPTION
**Pull request by leehart**

### CODE REVIEW

**Comment by leehart** [comment]
@alimanfoo I still need to resolve conflicts, test notebooks ("Community Edition" PyCharm doesn't have that, btw) and resolve any feedback.

@jorels1 mentioned that he might be able to get the ref genome seq annotations zarr produced for Af1 soon, so we might be able to squeeze that in too, if there's time.

**Comment by alimanfoo** [comment]
> test notebooks

Thanks Lee. To do this you can just do:

```
poetry run jupyter nbconvert --execute notebooks/*  --inplace
```

**Comment by alimanfoo** [comment]
> > test notebooks
> 
> Thanks Lee. To do this you can just do:
> 
> ```
> poetry run jupyter nbconvert --execute notebooks/*  --inplace
> ```

(This is what is run via github actions.)

File: malariagen_data/anopheles.py
Comment (alimanfoo) [inline malariagen_data/anopheles.py:50]: Suggest rename to "AnophelesDataResource" for naming consistency with plasmodium modules.

File: malariagen_data/af1.py
Comment (alimanfoo) [inline malariagen_data/af1.py:245]: Could this method be moved to the base class?

File: malariagen_data/af1.py
Comment (alimanfoo) [inline malariagen_data/af1.py:289]: Could this method be moved to the base class?

File: malariagen_data/af1.py
Comment (alimanfoo) [inline malariagen_data/af1.py:542]: There are several functions like this, where the implementation is fully generic and could be moved to the base class, but there is a parameter (in thise case `site_mask`) where the default value needs to be different in the child classes.

It would be good to move these to the base class somehow. There's several ways I could think of doing it. Perhaps the simplest would be to set the default value of the `site_mask` parameter to the string "default". Then the implementation could check if the value is "default" and replace it with the value of a class or instance attribute (e.g., `self._default_site_mask`) which is set in the child class.

File: malariagen_data/af1.py
Comment (alimanfoo) [inline malariagen_data/af1.py:747]: The taxon values "gcx1", "gcx2" and "gcx3" are only applicable to Ag3. 
```suggestion
    @staticmethod
    def _setup_taxon_colors(plot_kwargs):
        import plotly.express as px

        taxon_palette = px.colors.qualitative.Plotly
        taxon_color_map = {
            "funestus": taxon_palette[0],
        }
        plot_kwargs["color_discrete_map"] = taxon_color_map
        plot_kwargs["category_orders"] = {"taxon": list(taxon_color_map.keys())}
```

File: malariagen_data/af1.py
Comment (alimanfoo) [inline malariagen_data/af1.py:493]: Perhaps this could be a class variable, rather than a module variable? That would make it easier to move this function to the base class?

**Review by alimanfoo — COMMENTED** [review]
Hi @leehart, looking great, thanks so much for tackling this! I've looked through the implementations and direction all looks good. Just a couple of suggestions about opportunities to move some more code into the base class.

I'll review the tests separately.

**Review by alimanfoo — COMMENTED** [review]
Hi @leehart, just to say the approach for the unit tests LGTM, happy to go with this. There may be some opportunities to factor out more common tests in future but happy to improve that in future.

File: malariagen_data/af1.py
Comment (leehart) [reply malariagen_data/af1.py:747]: Of course, thanks! Will fix.

File: malariagen_data/af1.py
Comment (leehart) [reply malariagen_data/af1.py:245]: Yes, that shouldn't be there, thanks. Will fix.

File: malariagen_data/af1.py
Comment (leehart) [reply malariagen_data/af1.py:289]: The complication with generalising this one is the "special handling for 3.0 as data catalogs have a different format", but I'll take another look. I could move Af1's version to the parent class and have Ag3 override it. I suppose we can assume that Af1's version will/should be the standard, going forward.

Comment (leehart) [reply malariagen_data/af1.py:289]: ^ will do.

File: malariagen_data/af1.py
Comment (leehart) [reply malariagen_data/af1.py:542]: Yes, I agree, only I feel it might be slightly cleaner to default to `None` in the parent class, then check for `None` and replace that with the value an abstract property set by the child class, as you suggest. Thanks!

Comment (leehart) [reply malariagen_data/af1.py:542]: In the case of `plot_snps_track`, the subclasses actually have non-generic `cols` and `tooltips` too, but perhaps it will be OK to solve those in a similar way. (I wanted to avoid refactoring any functions that required non-trivial work to generalise, to avoid opening cans of worms and delaying the Af1 release, but now I appreciate that these cases are not very far off. Will do.)

File: malariagen_data/af1.py
Comment (leehart) [reply malariagen_data/af1.py:493]: Agreed, will do. Thanks!

**Comment by alimanfoo** [comment]
Hi @leehart, thanks so much for following up. In general happy to postpone any nontrivial refactoring, your call. I'll approve here and let you merge when you're happy to.

File: malariagen_data/af1.py
Comment (alimanfoo) [reply malariagen_data/af1.py:542]: > Yes, I agree, only I feel it might be slightly cleaner to default to `None` in the parent class, then check for `None` and replace that with the value an abstract property set by the child class, as you suggest. Thanks!

Not sure exactly what you mean here, but just flagging that we still need to allow for user to supply None meaning apply no site mask at all.

**Review by alimanfoo — APPROVED** [review]

File: malariagen_data/af1.py
Comment (leehart) [reply malariagen_data/af1.py:542]: That's a good point, thanks. (Sorry, I didn't mean to prompt any responses this week, so extra thanks!)

**Comment by leehart** [comment]
> Hi @leehart, thanks so much for following up. In general happy to postpone any nontrivial refactoring, your call. I'll approve here and let you merge when you're happy to.

Thanks @alimanfoo . Sorry I didn't meant to prompt you. I will be updating the PR shortly with resolved merge conflicts and additional changes. I believe that @cclarkson has kindly agreed to take a look in your stead.

File: malariagen_data/af1.py
Comment (leehart) [reply malariagen_data/af1.py:542]: FWIW, I'll use "default" as the default value instead of `None`, as you originally suggested, and I've checked that specifying `site_mask=None` when calling does indeed override the default of "default" to be `None`, rather than defaulting to "default" instead of `None`, if that makes sense.

**Comment by leehart** [comment]
It looks like the recently added `tests/test_af1.py::test_site_annotations` is failing for some reason. Will investigate/fix.

**Comment by leehart** [comment]
It looks like a memory issue (exit code 137).

**Comment by leehart** [comment]
OK, CI checks have passed, after temporarily avoiding the memory issue.  

@cclarkson -- It would be very reassuring to have your eyes on this too, please? Just because there has been quite a lot of reshuffling of the ~cabinet~ code, and I wouldn't want to break the ~economy~ package.

**Comment by cclarkson** [comment]
Hi @leehart, it all looks sensible to me. I haven't had a chance to run the code today, but if the tests are passing then I reckon it is good to go.

**Comment by leehart** [comment]
> Hi @leehart, it all looks sensible to me. I haven't had a chance to run the code today, but if the tests are passing then I reckon it is good to go.

OK, thanks @cclarkson . I'll bite the bullet and merge this in now.

**Pull request merged**
